### DISCUSSION TITLE
Forward run: best practices

### DISCUSSION THREAD

**Discussion by fjhheras** [Lightning-AI/pytorch-lightning#7298] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Once the model is trained, what would be the best way to use it to do a forward pass in a large dataset (that must be divided in batches not to fill GPU memory) and obtain the output?

What I normally do is to use the LightningModule as a torch Module. However, that forces me to: write a `for`, explicitly transfer to the device each batch, make sure to cal `.eval()`, concatenate the output of each batch... all the things that pytorch-lightning saves me from doing during training. But I guess I am missing an obvious simpler option?

**Comment by awaelchli** [comment, marked as answer]
In 1.3 there will be a [predict](https://pytorch-lightning.readthedocs.io/en/latest/common/trainer.html#predict) function that does all that for you!
You can already use it by installing 1.3.0rc1.

**Reply by fjhheras** [reply]
Cool to know! I will try to use the new version then
